sktime/transformers/rocket.py

```diff
@@ -0,0 +1,240 @@
+from numba import njit, prange
+import numpy as np
+import pandas as pd
+
+from sklearn.utils.validation import check_random_state
+from sktime.utils.validation import check_is_fitted
+
+from sktime.transformers.base import BaseTransformer
+from sktime.utils.data_container import dataframe_to_numpy
+from sktime.utils.validation.supervised import validate_X
+
+__author__ = "Angus Dempster"
+__all__ = ["ROCKET"]
+
+class Rocket(BaseTransformer):
+    """ROCKET
+
+    RandOm Convolutional KErnel Transform
+
+    @article{dempster_etal_2019,
+      author  = {Dempster, Angus and Petitjean, Fran\c{c}ois and Webb, Geoffrey I},
+      title   = {ROCKET: Exceptionally fast and accurate time series classification using random convolutional kernels},
+      year    = {2019},
+      journal = {arXiv:1910.13051}
+    }
+
+    Parameters
+    ----------
+    num_kernels : int, number of random convolutional kernels
+    seed        : int, random seed (optional, default None)
+    """
+
+    def __init__(self, num_kernels = 10_000, seed = None):
+        self.num_kernels = num_kernels
+        self.seed = seed
+
+    def fit(self, X):
+        """Infers time series length and number of channels / dimensions (for multivariate time series) from input pandas DataFrame, and generates random kernels.
+
+        Parameters
+        ----------
+        X : pandas DataFrame, input time series (sktime format)
+
+        Returns
+        -------
+        self
+        """
+        validate_X(X)
+        _, self.n_columns = X.shape
+        n_timepoints = X.applymap(lambda series : series.size).max().max()
+        self.kernels = _generate_kernels(n_timepoints, self.num_kernels, self.n_columns, self.seed)
+        self._is_fitted = True
+        return self
+
+    def transform(self, X, normalise = True):
+        """Transforms input time series using random convolutional kernels.
+
+        Parameters
+        ----------
+        X         : pandas DataFrame, input time series (sktime format)
+        normalise : boolean, whether or not to normalise the input time series per instance (default True)
+
+        Returns
+        -------
+        pandas DataFrame, transformed features
+        """
+        check_is_fitted(self, "_is_fitted")
+        validate_X(X)
+        _X = dataframe_to_numpy(X)
+        if normalise:
+            _X = (_X - _X.mean(axis = -1, keepdims = True)) / (_X.std(axis = -1, keepdims = True) + 1e-8)
+        return pd.DataFrame(_apply_kernels(_X, self.kernels))
+
+@njit("Tuple((float64[:],int32[:],float64[:],int32[:],int32[:],int32[:],int32[:]))(int64,int64,int64,optional(int64))")
+def _generate_kernels(n_timepoints, num_kernels, n_columns, seed):
+
+    if seed is not None:
+        np.random.seed(seed)

```

Comment:
Could we not use `check_random_state(random_state)` which returns a random state instance which you can then use to call the random functions, e.g.
```python
rng = check_random_state(random_state)
rng.choice(...)
```

Reply:
The NumPy `RandomState` (produced by `check_random_state`) can't be used inside Numba-compiled code.  Numba uses its own random state.   I've changed the code to accept an integer argument (seed), and ignore everything else.  This way it's compatible with the `random_state` parameter in other sktime code (i.e., won't throw an exception, etc.), but you are limited to using an integer seed.  I think that's the best I can do at this point.

Reply:
Okay, let's leave it as it is, from what I've read they're trying to make numpy and numba more consistent but still working on it.